port/cpl_float.h

```diff
@@ -49,13 +49,645 @@
 
 #include "cpl_port.h"
 
+#ifdef __cplusplus
+#include <algorithm>
+#include <cmath>
+#include <cstdint>
+#include <cstring>
+#include <limits>
+#ifdef HAVE_STD_FLOAT16_T
+#include <stdfloat>
+#endif
+#endif
+
 CPL_C_START
 GUInt32 CPL_DLL CPLHalfToFloat(GUInt16 iHalf);
 GUInt32 CPL_DLL CPLTripleToFloat(GUInt32 iTriple);
 CPL_C_END
 
 #ifdef __cplusplus
+
 GUInt16 CPL_DLL CPLFloatToHalf(GUInt32 iFloat32, bool &bHasWarned);
+
+GUInt16 CPL_DLL CPLConvertFloatToHalf(float fFloat32);
+float CPL_DLL CPLConvertHalfToFloat(GUInt16 nHalf);
+
+namespace cpl
+{
+
+// We define our own version of `std::numeric_limits` so that we can
+// specialized it for `CPLFloat16` if necessary. Specializing
+// `std::numeric_limits` doesn't always work because some libraries
+// use `std::numeric_limits`, and one cannot specialize a type
+// template after it has been used.
+template <typename T> struct CPLNumericLimits : std::numeric_limits<T>
```

Comment:
Prefixing names with `CPL` inside the `cpl` namespace seems redundant to me...

Reply:
We also have `cpl::CPLFloat16`.

I'd be happy to move it outside the namespace or to drop the prefix. Your choice?

Reply:
I'd defer to @rouault

Reply:
yes, let's drop the CPL prefix in both cases, and use cpl:: namespace

Reply:
Done.

Reply:
> Done.

hum, did you commit the changes, I still see cpl::CPLNumericLimits<>